test/unit/matrix/test_matrix.cpp

```diff
@@ -1423,6 +1580,23 @@ TEST(MatrixExceptionPropagation, RWPropagatesInRWAccess) {
   EXPECT_THROW(f.get(), CustomException);
 }
 
+TEST(MatrixExceptionPropagation, RWPropagatesInRWAccessWithSenderAdaptors) {
+  namespace ex = pika::execution::experimental;
+
+  auto matrix = createMatrix<TypeParam>();
+
+  auto s = matrix.readwrite_sender(LocalTileIndex(0, 0)) | ex::then([](auto&& f) {
+             // TODO: This reproduces the behaviour with futures, but is it a must?
+             auto f_local = std::move(f);
+             throw CustomException{};
+           }) |
+           ex::ensure_started();
```

Comment:
FYI, this is due to a difference in how/when stored values are released in `future::then`/`execution::then` (and all the other sender adaptors). I have to check if it's the `unwrapping` in the future-version that produces the difference in behaviour or if it's something else.

Reply:
It's `unwrapping` that makes the difference. I've added `unwrapping` to the sender tests as well and the behaviour is now the same.